### DISCUSSION TITLE
How to include data from a .shp file into my simulation

### DISCUSSION THREAD

**Discussion by martoconnh** [Parcels-code/Parcels#1447] [Q&A]
Hi! I have recently started using Parcels, and I am trying to recreate some simulations already done in MOHIDLagrangian by some coleagues. But I have been getting some doubts such as the following one:

I have created a fieldset using some .nc4 files (not any problems there). I also have a .shp file used in the previously mentioned MOHID simulation. So I was wondering whether there is any way I can include my shapefile in my Parcels simulation, and if so, how can I visualize it together with my fieldset in order to check if everything is okay?

Don't know if this is necessary but here it is the structure of my .shp file:

```
     id_poligon  p95 Hoja2_                                           geometry
0          3272          25  POLYGON ((-8.84472 42.62636, -8.84467 42.62665...
1          3273           1  POLYGON ((-8.80950 42.58337, -8.80911 42.58294...
2          3274           0  POLYGON ((-8.93103 42.44996, -8.93114 42.45078...
3          3275           0  POLYGON ((-8.82507 42.55814, -8.82497 42.55846...
4          3276          47  POLYGON ((-8.87916 42.60226, -8.87891 42.60286...
..          ...         ...                                                ...
468        3740           0  POLYGON ((-8.80314 42.61367, -8.80313 42.61366...
469        3741           0  POLYGON ((-8.80268 42.61362, -8.80268 42.61362...
470        3742           0  POLYGON ((-8.78646 42.59304, -8.78646 42.59305...
471        3743           0  POLYGON ((-8.78646 42.59290, -8.78645 42.59290...
472        3744           0  POLYGON ((-8.78665 42.59265, -8.78664 42.59265...

[473 rows x 3 columns]
```

**Comment by erikvansebille** [comment]
Dear @martoconnh, thank you for your question. However, we don't use shape files in Parcels. What do you use the shape file for? To set the starting locations of particles? Or something else?

**Reply by martoconnh** [reply]
Hi! So basically, my shape file contains the coast line of the region I am working with. This coast line is divided in some segments so that afterwards I can count how many particles from my simulation have arrived to each one. I am attaching images both of my whole coast line and a zoomed in picture where you can see the diferent segments.

Is there any way I can include any of this features in Parcels? Even if not using any shape file. I am sorry if this is some basic stuff but I have been having some trouble with it. Thank you so much for your help !

<img width="355" alt="coastline" src="https://github.com/OceanParcels/parcels/assets/147075668/5bef84b2-f1eb-4175-906e-9ef245866281">
<img width="275" alt="island_from_coastline" src="https://github.com/OceanParcels/parcels/assets/147075668/656ef4a8-2836-4a40-92c1-b230ed5e6855">

**Reply by michaeldenes** [reply]
Hi @martoconnh

Could you just do this in post-processing? Run your simulation as per normal, and afterwards run your analysis with your shapefiles?

An alternative may be to create a 'mask' from the shapefile (like a land mask, see the parcels tutorials for examples) that you can then add to the fieldset. This would allow you to check 'on-the-fly' if particles cross the coast using a custom kernel, but this would require a lot more work.

Good luck!
